userspace/libsinsp/filter.cpp

```diff
@@ -1677,6 +1677,30 @@ static void add_filtercheck_value(gen_event_filter_check *chk, size_t idx, const
 	}
 }
 
+static void add_evttypes_names(gen_event_filter_check *chk, const std::string& value, std::unordered_set<std::string>& m_filter_evttypes_names)
+{
+	// TODO fix: Currently includes negative evt.type checks e.g. "and not evt.type in (connect, accept)"
```

Comment:
/cc @jasondellaluce 
Is there a better way to achieve this?

Reply:
I think we should be working on ASTs to achieve this like we do on Falco (https://github.com/falcosecurity/falco/blob/master/userspace/engine/filter_evttype_resolver.h). We may eventually consider extending that Falco utility class to also grab the names and pushing it down to libsinsp.

Reply:
Totally agree :+1:

Reply:
Agree, following suggestion re evttypes extraction from filters:

1. Open a Falco PR, collaborate with @jasondellaluce on it (some more thoughts to follow in another comment shortly)
2. After addressing the actual Falco use case and gaining a better understanding of the "AST" parts of the code base, come back here and see what the best approach could be to also support `sinsp-example` that only has one simple filter condition in order to be able to support additional `e2e` tests.